### PR TITLE
csi: reap unused volume claims at leadership transitions

### PR DESCRIPTION
**Pull request by tgross**
When `volumewatcher.Watcher` starts on the leader, it starts a watch
on every volume and triggers a reap of unused claims on any change to
that volume. But if a reaping is in-flight during leadership
transitions, it will fail and the event that triggered the reap will
be dropped. Perform one reap of unused claims at the start of the
watcher so that leadership transitions don't drop this event.

This bug contributes to https://github.com/hashicorp/nomad/issues/10927 but is not the full cause of all cases.

### CODE REVIEW

File: nomad/volumewatcher/volumes_watcher_test.go [hunk 1]
```diff
@@ -52,8 +52,8 @@ func TestVolumeWatch_EnableDisable(t *testing.T) {
 	require.Equal(0, len(watcher.watchers))
 }
 
-// TestVolumeWatch_Checkpoint tests the checkpointing of progress across
-// leader leader step-up/step-down
+// TestVolumeWatch_LeadershipTransition tests the correct behavior of
+// claim reaping across leader step-up/step-down
 func TestVolumeWatch_Checkpoint(t *testing.T) {
```
Comment (jrasell) [inline nomad/volumewatcher/volumes_watcher_test.go:57]: Should we change the function name to match the comment, or visa versa?

**Review by jrasell — APPROVED** [review]
Minor comment about the test func name but not a blocker and absolutely LGTM!

File: nomad/volumewatcher/volumes_watcher_test.go (hunk 1 above)
Comment (tgross) [reply nomad/volumewatcher/volumes_watcher_test.go:57]: Oops, thanks for catching that.

**Comment by tgross** [comment]
This was released in 1.2.4 but I missed backporting it to 1.1.x and 1.0.x. I'll backport it as part of 1.2.5.

**Pull request merged**